chapter9/src/9_5_3/main.rs

```diff
@@ -0,0 +1,31 @@
+use std::env;
+use std::process::exit;
+
+// whichコマンド
+// PATH環境変数のパス一覧を取得してきて、split_pathsで分割
+// その後、各パスの下に最初の引数で指定された実行ファイルがあるかをチェック
+fn main() {
+    let args: Vec<String> = env::args().collect();
+    if args.len() == 1 {
+        println!("{} [exec file name]", args[0]);
+        exit(1);
+    }
+
+    match env::var_os("PATH") {
+        Some(paths) => {
+            for path in env::split_paths(&paths) {
+                let exec_path = path.join(&args[1]);
+                if exec_path.exists() {
+                    println!("{}", exec_path.to_str().expect("faled to convert to str"));
+                    return;
+                }
+            }
+        }
+        None => {
+            println!("failed to get PATH");
+            exit(1)
+        }
+    }
+
+    exit(1);
```

Comment:
上記2つのコメントを踏まえるとこんな感じにかけます。

```suggestion
    let paths = env::var_os("PATH").expect("failed to get PATH");
    let opt_exec_path = env::split_paths(&paths).find_map(|path| {
        let exec_path = path.join(&args[1]);
        if exec_path.exists() {
            Some(exec_path)
        } else {
            None
        }
    });
    if let Some(exec_path) = opt_exec_path {
        println!("{}", exec_path.to_str().expect("faled to convert to str"));
    } else {
        exit(1);
    }

```

Reply:
ありがとうございます！
確かにパターンマッチの分岐で型が合わないのが違和感だったので、イテレータを利用したほうがスッキリして良いですね！
ご提案頂いた内容で修正致しました。🙏